pkg/proxy/logger/logger.go

```diff
@@ -220,7 +220,9 @@ func (lr *LogRecord) getLogFields() *logrus.Entry {
 
 // Log logs a record to the logfile and flushes the buffer
 func (lr *LogRecord) Log() {
-	flowdebug.Log(lr.getLogFields(), "Logging flow record")
+	flowdebug.Log(func() (*logrus.Entry, string) {
+		return lr.getLogFields(), "Logging flow record"
```

Comment:
I was taking a look at the method `getLogFields` and I think it can be optimized a bit if you'd like to roll that into this PR. The method `getLogFields` makes, at worse, two calls to `WithFields` and we could collapse it into one:

```patch
diff --git a/pkg/proxy/logger/logger.go b/pkg/proxy/logger/logger.go
index 9fdd237a79..a31f4cde08 100644
--- a/pkg/proxy/logger/logger.go
+++ b/pkg/proxy/logger/logger.go
@@ -190,32 +190,28 @@ func (lr *LogRecord) ApplyTags(tags ...LogTag) {
 }
 
 func (lr *LogRecord) getLogFields() *logrus.Entry {
-       fields := log.WithFields(logrus.Fields{
+       fields := logrus.Fields{
                FieldType:    lr.Type,
                FieldVerdict: lr.Verdict,
                FieldMessage: lr.Info,
-       })
+       }
 
        if lr.HTTP != nil {
-               fields = fields.WithFields(logrus.Fields{
-                       FieldCode:     lr.HTTP.Code,
-                       FieldMethod:   lr.HTTP.Method,
-                       FieldURL:      lr.HTTP.URL,
-                       FieldProtocol: lr.HTTP.Protocol,
-                       FieldHeader:   lr.HTTP.Headers,
-               })
+               fields[FieldCode] = lr.HTTP.Code
+               fields[FieldMethod] = lr.HTTP.Method
+               fields[FieldURL] = lr.HTTP.URL
+               fields[FieldProtocol] = lr.HTTP.Protocol
+               fields[FieldHeader] = lr.HTTP.Headers
        }
 
        if lr.Kafka != nil {
-               fields = fields.WithFields(logrus.Fields{
-                       FieldCode:               lr.Kafka.ErrorCode,
-                       FieldKafkaAPIKey:        lr.Kafka.APIKey,
-                       FieldKafkaAPIVersion:    lr.Kafka.APIVersion,
-                       FieldKafkaCorrelationID: lr.Kafka.CorrelationID,
-               })
+               fields[FieldCode] = lr.Kafka.ErrorCode
+               fields[FieldKafkaAPIKey] = lr.Kafka.APIKey
+               fields[FieldKafkaAPIVersion] = lr.Kafka.APIVersion
+               fields[FieldKafkaCorrelationID] = lr.Kafka.CorrelationID
        }
 
-       return fields
+       return log.WithFields(fields)
 }
 
 // Log logs a record to the logfile and flushes the buffer
```

Reply:
> The method getLogFields makes, at worse, two calls to WithFields

Correct as `lr.Kafka != nil` and `lr.HTTP != nil` should be mutually exclusive, thus maybe we could refactor in a `switch` statement (or `if` / `else if`) to make it clearer.

Ryan's suggestion also will not trigger a map grow since we're at most setting 8 key/val in the HTTP case, maybe we could explicitly initialize the map with the known max key/val set by `getLogFields`:

```
// NOTE: we're adding at most 8 key/val from this func, allocate enough space to avoid having to grow the map.
fields := make(logrus.Fields, 8)
fields[FieldType] = lr.Type
fields[FieldVerdict] = lr.Verdict
fields[FieldMessage] = lr.Info
```

Reply:
Applied both of these suggestions, thanks!